src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2871,29 +2902,67 @@ def add(attr):
                 "method": "pool.dataset.permission",
                 "params": ["tank/myuser", {
                     "user": "myuser",
+                    "acl": [],
                     "group": "wheel",
                     "mode": "755",
-                    "recursive": true,
+                    "options": {"recursive": true, "stripacl": true},
                 }]
             }
+
         """
         path = (await self._get_instance(id))['mountpoint']
         user = data.get('user', None)
         group = data.get('group', None)
+        uid = gid = -1
         mode = data.get('mode', None)
-        recursive = data.get('recursive', False)
-        acl = data['acl']
+        options = data.get('options', {})
+        acl = data.get('acl', [])
+
         verrors = ValidationErrors()
+        if user:
+            try:
+                uid = (await self.middleware.call('dscache.get_uncached_user', user))['pw_uid']
+            except Exception as e:
+                verrors.add('pool_dataset_permission.user', str(e))
 
-        if (acl == 'UNIX' or acl == 'MAC') and mode is None:
+        if group:
+            try:
+                gid = (await self.middleware.call('dscache.get_uncached_group', group))['gr_gid']
+            except Exception as e:
+                verrors.add('pool_dataset_permission.group', str(e))
+
+        if acl and mode:
             verrors.add('pool_dataset_permission.mode',
-                        'This field is required')
+                        'setting mode and ACL simultaneously is not permitted.')
 
         if verrors:
             raise verrors
 
-        await self.middleware.call('notifier.mp_change_permission', path, user,
-                                   group, mode, recursive, acl.lower())
+        if not acl and mode is None:
+            await run(
+                '/usr/sbin/chown',
+                '-R' if not options['traverse'] else '-Rx',
+                f'{uid}:{gid}', path, check=False
+            )
```

Comment:
setacl and setperm are jobs and wont block permission call.

chown on the other hand may take a long time and block this call. Should we use Popen and run `communicate` in a background task?